.github/workflows/merge.yml

```diff
@@ -409,34 +440,93 @@ jobs:
+          echo "Total Num of local registers is $NUM_OF_REGISTERS_first and in vault is $NUM_OF_REGISTERS_IN_VAULT"
           
           rm -rf file_list.txt register_list.txt vault_data.txt
 
           python3 -c 'import sys; assert sys.argv[1] == sys.argv[2], f"Error: Local public Files: {sys.argv[1]} and vault public files: {sys.argv[2]} are Not Equal"' $NUM_OF_PUBLIC_FILES $NUM_OF_PUBLIC_FILES_IN_VAULT
           python3 -c 'import sys; assert sys.argv[1] == sys.argv[2], f"Error: Local private Files: {sys.argv[1]} and vault private files: {sys.argv[2]} are Not Equal"' $NUM_OF_PRIVATE_FILES $NUM_OF_PRIVATE_FILES_IN_VAULT
-          # python3 -c 'import sys; assert sys.argv[1] == sys.argv[2], f"Error: Local registers: {sys.argv[1]} and vault registers: {sys.argv[2]} are Not Equal"' $NUM_OF_REGISTERS $NUM_OF_REGISTERS_IN_VAULT
+          python3 -c 'import sys; assert sys.argv[1] == sys.argv[2], f"Error: Local registers: {sys.argv[1]} and vault registers: {sys.argv[2]} are Not Equal"' $NUM_OF_REGISTERS_first $NUM_OF_REGISTERS_IN_VAULT
           echo "vault synced successfully!"
         env:
           SN_LOG: "v"
         timeout-minutes: 15
 
+      - name: Set up variables - vault sync - windows
+        if: matrix.os == 'windows-latest'
+        shell: pwsh
+        run: |
+          $ErrorActionPreference = "Stop"
+          ./target/release/autonomi --log-output-dest=data-dir file list > file_list.txt 2>&1
+          ./target/release/autonomi register list > register_list.txt 2>&1
+          ./target/release/autonomi --log-output-dest=data-dir vault load > vault_data.txt 2>&1
+        env:
+          SN_LOG: "v"
+        timeout-minutes: 15
+
+      - name: Vault sync validation
+        if: matrix.os == 'windows-latest'
+        shell: python
+        run: |
+          import re
+          def find_number_before_word(file_name, search_word):
+              """
+              Reads a file and finds the number immediately preceding a specified word in a line.
+
+              :param file_name: Name of the file to read.
+              :param search_word: Word to search for in the file.
+              :return: The number before the word as an integer, or None if not found.
+              """
+              try:
+                  with open(file_name, 'r') as file:
+                      for line in file:
+                          if search_word in line:
+                              match = re.search(r'(\d+)\s+' + re.escape(search_word), line)
+                              if match:
+                                  return int(match.group(1))  # Convert to integer
+                  return None  # Return None if no match is found
+              except FileNotFoundError:
+                  print(f"Error: File '{file_name}' not found.")
+                  return None
+          NUM_OF_PUBLIC_FILES = find_number_before_word("file_list.txt", "public")
+          print("NUM_OF_PUBLIC_FILES:", NUM_OF_PUBLIC_FILES)
```

Comment:
Can we not directly use `rg` here instead of using Python?
Assuming the file content is something like the one below, then rg should suffice.
```
❯ echo "1000 public\n2000 private" | rg '\d+\s+public' | awk '{print $1}'
1000
```

Reply:
This is not working and python makes it easier to read from the file and make a comparison operation within the one job in windows.